### PR TITLE
fix: tracer setup

### PR DESCRIPTION
**Pull request by InoMurko**
/

## Overview

As reported in the devops issue, Child Chain Phoenix has been taking Watcher's Datadog tracer causing crashes.

## Changes

- Set the correct tracer per application mode in multiple places.

## Testing
/

### CODE REVIEW

**Comment by InoMurko** [comment]
I'm merging this into v0.2 @pdobacz is that ok?

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-0.2.2
+0.2.3
```
Comment (pdobacz) [inline VERSION]: rebase on the HEAD of v0.2, this already got bumped

File: apps/omg_child_chain_rpc/lib/omg_child_chain_rpc/application.ex [hunk 2]
```diff
@@ -22,7 +22,8 @@ defmodule OMG.ChildChainRPC.Application do
 
   def start(_type, _args) do
     DeferredConfig.populate(:omg_child_chain_rpc)
-
+    DeferredConfig.populate(:omg_watcher_rpc)
```
Comment (pdobacz) [inline apps/omg_child_chain_rpc/lib/omg_child_chain_rpc/application.ex]: `omg_watcher_rpc` in chch?

and the same vice versa in `omg_watcher_rpc`. If those are intended, a comment would be v. useful

File: apps/omg_watcher_rpc/lib/application.ex [hunk 3]
```diff
@@ -19,7 +19,8 @@ defmodule OMG.WatcherRPC.Application do
 
   def start(_type, _args) do
     DeferredConfig.populate(:omg_watcher_rpc)
-
+    DeferredConfig.populate(:omg_child_chain_rpc)
+    :ok = Application.put_env(:spandex_phoenix, :tracer, OMG.WatcherRPC.Tracer, persistent: true)
```
Comment (pdobacz) [inline apps/omg_watcher_rpc/lib/application.ex]: oh my, now I see, ouch. So `spandex_phoenix` is overwritten for every umbrella subapp, right?

If there isn't any other way to mend this, comments would be v. useful.

File: apps/omg_child_chain_rpc/lib/omg_child_chain_rpc/application.ex (hunk 2 above)
Comment (InoMurko) [reply apps/omg_child_chain_rpc/lib/omg_child_chain_rpc/application.ex]: yeah, that's kinda the issue we were seeing, tracers are global settings that don't support more then one phoenix application, so watcher's configuration comes in first and :spandex_datadog picks that one.

File: apps/omg_watcher_rpc/lib/application.ex (hunk 3 above)
Comment (InoMurko) [reply apps/omg_watcher_rpc/lib/application.ex]: will do!

**Comment by InoMurko** [comment]
> LGTM with some comments. Will you be cherry-picking this to `master` as well?

master does not have this problem, because of configuration providers that sort this out before any of the (dependency) applications boot.

**Pull request merged**